curator-x-discovery/src/test/java/org/apache/curator/x/discovery/TestServiceDiscovery.java

```diff
@@ -297,4 +304,70 @@ public void testNoServerOnStart() throws Exception
             }
         }
     }
+
+    // CURATOR-164
+    @Test
+    public void testUnregisterService() throws Exception {
+        final String name = "name";
+
+        final CountDownLatch restartLatch = new CountDownLatch(1);
+        final CountDownLatch serializeLatch = new CountDownLatch(1);
```

Comment:
you made this latch, but then used a sleep call to wait for the serialization to finish. better ot switch to using this latch.